gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/PRHARedundancyProvider.java

```diff
@@ -1159,6 +1144,12 @@ public boolean createBackupBucketOnMember(final int bucketId,
           return false;
         }
       }
+      catch (VirtualMachineError err) {
+        SystemFailure.initiateFailure(err);
+        // If this ever returns, rethrow the error.  We're poisoned
+        // now, so don't let this thread continue.
+        throw err;
+      }
       catch (Throwable e) {
```

Comment:
Get rid of these blocks. We deliberately changed this to "catch (Throwable e) { if (e instanceof Error && SystemFailure.isJVMFailureError(err)) }" because former also initiates failure for errors like StackOverflowError or ZipError which simply indicates a code/data bug rather than a fatal JVM error. The isJVMFailure restricts the check to mostly OOME.